go-controller/pkg/ovn/ovn.go

```diff
@@ -494,7 +498,57 @@ func (oc *Controller) WatchPods() {
 	oc.WatchResource(oc.retryPods)
 }
 
-// WatchNetworkPolicy starts the watching of the network policy resource and calls
+// now that we have added all the OVN ACLs with optimization, it is time to remove the stale OVN
+// ACL entries from the database
+func (oc *Controller) CleanStaleNetworkPolicy() {
+	start := time.Now()
+	klog.V(5).Infof("Cleaning up stale OVN ACLs that are left behind after L4 Port consolidation")
+
+	// want ACLs configured that don't have l4fused key and that have l4Match set (but not to None)
+	pACL := func(item *nbdb.ACL) bool {
+		if _, ok := item.ExternalIDs[l4MatchFusedExtIdKey]; !ok {
+			if val, ok := item.ExternalIDs[l4MatchACLExtIdKey]; ok {
+				if val != noneMatch {
+					return true
+				}
+			}
+		}
+		return false
+	}
+
+	staleACLs, err := libovsdbops.FindACLsWithPredicate(oc.nbClient, pACL)
+	if err != nil {
+		klog.Warningf("Failed to retrieve stale OVN ACL entries that wre not optimized " +
+			"for L4 Ports consolidation: %v, err")
+	} else {
+		// it could be that delete all the acls in one go might fail for various reasons,
+		// so lets try to delete one at a time so that we can remove as many stale acls
+		// as possible.
+		klog.V(5).Infof("Number of staleACLS to be cleaned is %d", len(staleACLs))
+		for _, staleACL := range staleACLs {
+			staleACL := staleACL
+			nsName := staleACL.ExternalIDs[namespaceACLExtIdKey]
+			policyName := staleACL.ExternalIDs[policyACLExtIdKey]
+			pgName := fmt.Sprintf("%s_%s", nsName, policyName)
+			pgName = hashedPortGroup(pgName)
+			aclDesc := fmt.Sprintf("stale ACL %s/%s/%s in port group %s", staleACL.UUID, nsName, policyName, pgName)
+			klog.V(5).Infof("About to delete %s", aclDesc)
+			ops, err := libovsdbops.DeleteACLsFromPortGroupOps(oc.nbClient, nil, pgName, staleACL)
+			if err != nil {
+				klog.Warningf("Failed to get ops to delete %s: %v", aclDesc, err)
+				continue
+			}
+			_, err = libovsdbops.TransactAndCheck(oc.nbClient, ops)
+			if err != nil {
+				klog.Warningf("Failed to delete %s: %v", aclDesc, err)
+				continue
+			}
+		}
+	}
+	klog.V(5).Infof("Completed cleaning up stale OVN ACLs in %v", time.Since(start))
```

Comment:
just defer this at the top of the function, then you can get rid of the else above and just return on error.

Reply:
I see we are already printing how many stale ACLs we found on line 527. Should we show in this deferred print how many of these stale ACLs we actually managed to delete?